### PR TITLE
ci: use pkgconfig as a fallback for finding SDL2_ttf

### PR DESCRIPTION
**Pull request by hch12907**

### CODE REVIEW

**Comment by hrydgard** [comment]
```
/home/runner/work/ppsspp/ppsspp/Common/Render/Text/draw_text_sdl.cpp:382:33: error: use of undeclared identifier 'TTF_WRAPPED_ALIGN_CENTER'
                TTF_SetFontWrappedAlign(font, TTF_WRAPPED_ALIGN_CENTER);
```

Hm, that's a weird one, are we getting an old version or something?

**Comment by hch12907** [comment]
Apparently text alignment came with the same release that added Cmake support...

 https://github.com/libsdl-org/SDL_ttf/releases/tag/release-2.20.0

**Comment by hch12907** [comment]
Yeah, #ifdef-ing out the text alignment code seems to work. I hope to eventually revert this PR when the Ubuntu CI gets updated though...

**Comment by hrydgard** [comment]
This seems to work on my machine.

I'm going to merge this after tagging 1.16.2, so that even if this causes problems with the flatpak buildbot somehow, we'll get a good build in between.

**Review by hrydgard — APPROVED** [review]

**Comment by Kelvfimer** [comment]
Hello

I'm getting this error message when building due to this PR. PLs can you take a look into it? thx

[111/919] Building CXX object CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o
FAILED: CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o

PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp:146:22: error: 'TTF_GlyphIsProvided32' was not declared in this scope; did you mean 'TTF_GlyphIsProvided'?

Platform
----------------------------------------
Linux / BSD

Compiler and build tool versions
----------------------------------------
CMAKE 3.24.2 gcc 12.2.0

Operating system version
----------------------------------------
ubuntu 22.04 WSL2 via toolchain Emuelec 4.6 to target board Odroid n2+

Build commands used
----------------------------------------
PROJECT=Amlogic-ce DEVICE=Amlogic-ng ARCH=aarch64 DISTRO=EmuELEC ./scripts/build ppsspp
PROJECT=Amlogic-ce DEVICE=Amlogic-ng ARCH=aarch64 DISTRO=EmuELEC ./scripts/build PPSSPPSDL

What happens 
--------------------------------------
When compiling this pull request that includes commit 3aed81d51bfc024d0c354d50296c7273c69c8325 I'm getting 

[111/919] Building CXX object CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o
FAILED: CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o

PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp:146:22: error: 'TTF_GlyphIsProvided32' was not declared in this scope; did you mean 'TTF_GlyphIsProvided'?

Last Working Commit
--------------------------
https://github.com/hrydgard/ppsspp/commit/50da8a91da54da808c6b1348ba445d2cb2529f9e

Issue started on commit
--------------------------
https://github.com/hrydgard/ppsspp/commit/3aed81d51bfc024d0c354d50296c7273c69c8325

Full log
--------------------------
[111/919] Building CXX object CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o
FAILED: CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o
/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/toolchain/bin/ccache /home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/toolchain/bin/aarch64-libreelec-linux-gnu-g++ -DASSETS_DIR="/usr/share/ppsspp/assets/" -DEGL_NO_X11 -DHTTPS_NOT_AVAILABLE -DMINIUPNPC_SET_SOCKET_TIMEOUT -DMINIUPNP_STATICLIB -DPNG_ARM_NEON_OPT=0 -DSDL -DSHARED_ZLIB -DSTACK_LINE_READER_BUFFER_SIZE=1024 -DUSE_SDL2_TTF -DUSE_SDL2_TTF_FONTCONFIG -DUSE_SDL2_TTF_PKGCONFIG -DUSING_FBDEV -DUSING_GLES2 -DWITH_UPNP -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE=1 -D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=700 -D_XOPEN_SOURCE_EXTENDED -D__BSD_VISIBLE=1 -D__STDC_CONSTANT_MACROS -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/glslang -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/glew -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053 -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/openxr -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/libzip -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/libpng17 -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/zstd/lib -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/.aarch64-libreelec-linux-gnu -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/snappy/. -I/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/ext/cmake/cpu_features/../../cpu_features/include -march=armv8-a+crc+fp+simd -mabi=lp64 -Wno-psabi -mtune=cortex-a73.cortex-a53 -mno-outline-atomics -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -DDUMMYOPT -Os -D_NDEBUG -std=gnu++17 -Wformat -Wno-multichar -fno-math-errno -Wno-psabi -fPIC -fno-strict-aliasing -MD -MT CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o -MF CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o.d -o CMakeFiles/Common.dir/Common/Render/Text/draw_text_sdl.cpp.o -c /home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp
/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp: In member function 'void TextDrawerSDL::PrepareFallbackFonts()':
/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp:61:27: warning: comparison of integer expressions of different signedness: 'int' and 'long unsigned int' [-Wsign-compare]
61 | for (int i = 0; i < ARRAY_SIZE(names); i++) {
/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp: In member function 'uint32_t TextDrawerSDL::CheckMissingGlyph(const std::string&)':
/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp:142:27: warning: comparison of integer expressions of different signedness: 'int' and 'std::__cxx11::basic_string::size_type' {aka 'long unsigned int'} [-Wsign-compare]
142 | for (int i = 0; i < text.length(); ) {
| ~~^~~~~~~~~~~~~~~
/home/kelv/EmuELEC/build.EmuELEC-Amlogic-ng.aarch64-4/build/PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp:146:22: error: 'TTF_GlyphIsProvided32' was not declared in this scope; did you mean 'TTF_GlyphIsProvided'?
146 | if (!TTF_GlyphIsProvided32(font, glyph)) {

PPSSPPSDL-a9b6421dfde68be9d065b51e9d519699ce2e9053/Common/Render/Text/draw_text_sdl.cpp:146:22: error: 'TTF_GlyphIsProvided32' was not declared in this scope; did you mean 'TTF_GlyphIsProvided'?

Pls can you double check it?

thx

**Comment by hch12907** [comment]
You need a newer version of SDL2_ttf. Something greater or equal to [2.0.18](https://github.com/libsdl-org/SDL_ttf/releases/tag/release-2.0.18).

As a workaround, change this line:
```c
if (!TTF_GlyphIsProvided32(font, glyph)) {
```

to this:
```c
if (!TTF_GlyphIsProvided(font, glyph)) {
```

**Comment by Kelvfimer** [comment]
No worries I'll try with a newer version of SDL2_ttf. If not working I'll Pr the change you told.

Thx @hch12907

File: Common/Render/Text/draw_text_sdl.cpp [hunk 1]
```diff
@@ -374,12 +374,14 @@ void TextDrawerSDL::DrawStringBitmap(std::vector<uint8_t> &bitmapData, TextStrin
 		font = fallbackFonts_[0];
 	}
 
+#ifndef USE_SDL2_TTF_PKGCONFIG
```
Comment (unknownbrackets) [inline Common/Render/Text/draw_text_sdl.cpp:377]: This should instead use `#if SDL_TTF_VERSION_ATLEAST(...)`.

-[Unknown]

**Comment by unknownbrackets** [comment]
> You need a newer version of SDL2_ttf. Something greater or equal to [2.0.18](https://github.com/libsdl-org/SDL_ttf/releases/tag/release-2.0.18).

That's another great case for using `SDL_TTF_VERSION_ATLEAST(2, 0, 18)`.

-[Unknown]

**Comment by Kelvfimer** [comment]
@unknownbrackets @hch12907 thx for your help, I upgraded to SDL_TFF 2.0.18 and the issue was solved :)

**Pull request merged**